src/tools/pub/glossary.md

```diff
@@ -29,6 +29,74 @@ specify `any` for their dependencies' [version constraints][].
 [lockfiles]: #lockfile
 [version constraints]: #version-constraint
 
+## Content hashes
+
+The pub.dev repository maintains a sha256 hash of each package version it hosts.
+The pub client can use this hash to validate the integrity of downloaded packages,
+and protect against changes on the repository. 
+
+When `dart pub get` downloads a package, it computes the hash of the
+downloaded archive. The hash of each hosted dependency is stored with the
+[resolution][] in the [lockfile][].
+
+The pub client uses this content hash to verify that running `pub get` again using
+the same lockfile, potentially on a different computer, uses exactly the same packages.
```

Comment:
Maybe "exactly the same package" could be made a bit more specific, since it could protect from matching packages, just modified or outdated.

I know pub uses the term ["package archive"](https://github.com/dart-lang/pub/blob/master/doc/repository-spec-v2.md#list-all-versions-of-a-package) to describe the saved contents of the file, but perhaps that would sound confusing here. Any thoughts?

I'll think about it some more, but if no one has ideas, I suppose we can just leave this as is.

Reply:
Maybe "uses the same packages in the same state", or "uses the correct package states"?

I don't really understand the topic enough to give a better suggestion